polkadot/node/network/protocol/src/grid_topology.rs

```diff
@@ -89,6 +89,25 @@ impl SessionGridTopology {
 		SessionGridTopology { shuffled_indices, canonical_shuffling, peer_ids }
 	}
 
+	/// Updates the known peer ids for the passed authorithies ids.
+	pub fn update_authority_ids(
+		&mut self,
+		peer_id: PeerId,
+		ids: &HashSet<AuthorityDiscoveryId>,
+	) -> bool {
+		if self.peer_ids.contains(&peer_id) {
+			for peer in self
+				.canonical_shuffling
+				.iter_mut()
+				.filter(|peer| ids.contains(&peer.discovery_id))
+			{
+				peer.peer_ids.push(peer_id);
+				self.peer_ids.insert(peer_id);
```

Comment:
If it is already in peer_ids, why do we insert it?

Reply:
Yeah, that's why `tests` it is on the todo list.

I was looking more about what you think of the approach does it seem reasonable to you ?

Reply:
Seems reasonable at first sight, yes.